src/mol-model-props/computed/membrane-orientation.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2019-2020 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Sebastian Bittrich <sebastian.bittrich@rcsb.org>
+ * @author Alexander Rose <alexander.rose@weirdbyte.de>
+ */
+
+import { ParamDefinition as PD } from '../../mol-util/param-definition';
+import { Structure } from '../../mol-model/structure';
+import { CustomStructureProperty } from '../common/custom-structure-property';
+import { CustomProperty } from '../common/custom-property';
+import { CustomPropertyDescriptor } from '../../mol-model/custom-property';
+import { ANVILParams, ANVILProps, computeANVIL } from './membrane-orientation/ANVIL';
+import { AccessibleSurfaceAreaProvider } from './accessible-surface-area';
+import { MembraneOrientation } from '../../mol-model/structure/model/properties/membrane-orientation';
+
+function getMembraneOrientationParams(data?: Structure) {
+    let defaultType = 'anvil' as 'anvil' | 'db'; // TODO flip - OPM (or some other db-source) should be default if PDB identifier is known
```

Comment:
see https://github.com/molstar/molstar/blob/master/src/mol-model-props/computed/secondary-structure.ts#L18-L38 for example code

Reply:
Is there an example that shows how parameters can be passed to this kind of provider? Along the lines of: `await MembraneOrientationProvider.attach(ctx, structure, ...);` The signature calls for `Mapped` and `NamedParams` but I struggle to construct the correct parameter object.

Reply:
for

```js
{
        type: PD.MappedStatic(defaultType, {
            'model': PD.EmptyGroup({ label: 'Model' }),
            'dssp': PD.Group(DSSPComputationParams, { label: 'DSSP', isFlat: true })
        }, { options: [['model', 'Model'], ['dssp', 'DSSP']] })
}
```

it is

```js
{ type: { name: 'model', params: {} } }
```

and

```js
{ type: { name: 'dssp', params: { oldDefinition: true, oldOrdering: true } } }
```

Reply:
Oh, I wanna say duh. Didn't realize that it's so straight forward to pass the params in. Thanks!